tools/code-owners-parser/Azure.Sdk.Tools.RetrieveCodeOwners/Program.cs

```diff
@@ -10,26 +11,32 @@ class Program
         /// Retrieves codeowners information for specific section of the repo
         /// </summary>
         /// <param name="targetDirectory">The directory whose information is to be retrieved</param>
-        /// <param name="rootDirectory">The root of the repo or $(Build.SourcesDirectory) on DevOps</param>
-        /// <param name="vsoOwningUsers">Variable for setting user aliases</param>
-        /// <returns></returns>
+        /// <param name="codeOwnerFilePath">The path of CODEOWNERS file in repo</param>
+        /// <returns>Exit code</returns>
 
-        public static void Main(
+        public static int Main(
             string targetDirectory,
-            string rootDirectory,
-            string vsoOwningUsers
+            string codeOwnerFilePath
             )
         {
             var target = targetDirectory.ToLower().Trim();
-            var codeOwnersLocation = Path.Join(rootDirectory, ".github", "CODEOWNERS");
-            var owners = CodeOwnersFile.ParseAndFindOwnersForClosestMatch(codeOwnersLocation, target);
-            if (owners == null)
-            {
-                Console.WriteLine(String.Format("We cannot find any closest code owners from the target path {0}", targetDirectory));
+            try {
+                var codeOwnerEntry = CodeOwnersFile.ParseAndFindOwnersForClosestMatch(codeOwnerFilePath, target);
+                if (codeOwnerEntry == null)
+                {
+                    Console.Error.WriteLine(String.Format("We cannot find any closest code owners from the target path {0}", targetDirectory));
+                    return 1;
+                }
+                else
+                {
+                    var codeOwnerJson = JsonSerializer.Serialize<CodeOwnerEntry>(codeOwnerEntry);
```

Comment:
Do you need to do something like below to get pretty printing of the json, or is it already handled?

```suggestion
                        var codeOwnerJson = JsonSerializer.Serialize<CodeOwnerEntry>(codeOwnerEntry, new JsonSerializerOptions { WriteIndented = true });
```

Reply:
Nice to have.